### PR TITLE
Add integration test for jsonb comparison with boolean values

### PR DESCRIPTION
**Pull request by leandrocp**
Follow up of https://github.com/elixir-ecto/ecto_sql/pull/399

### CODE REVIEW

**Comment by leandrocp** [comment]
@josevalim what's the approach to deal with

```
   +integration-test *failed* |   1) test typed boolean map (Ecto.Integration.TypeTest)
   +integration-test *failed* |      /src/ecto/integration_test/cases/type.exs:256
   +integration-test *failed* |      ** (Ecto.QueryError) Tds adapter does not support json_extract_path expression, use fragment with JSON_VALUE/JSON_QUERY in query:
   +integration-test *failed* |      
   +integration-test *failed* |      from p0 in Ecto.Integration.Post,
   +integration-test *failed* |        where: p0.meta["enabled"] == true,
   +integration-test *failed* |        select: p0.meta
   ```
   
   ?

File: integration_test/cases/type.exs [hunk 1]
```diff
@@ -252,6 +252,18 @@ defmodule Ecto.Integration.TypeTest do
            [%{"foo" => 1.0, "bar" => 416500.0}]
   end
 
+  @tag :map_type
```
Comment (wojtekmach) [inline integration_test/cases/type.exs:255]: please add this tag so it's excluded on TDS

```suggestion
  @tag :map_type
  @tag :json_extract_path
```

File: integration_test/cases/type.exs [hunk 2]
```diff
@@ -307,12 +318,16 @@ defmodule Ecto.Integration.TypeTest do
     assert TestRepo.one(from o in Order, select: o.meta["'single quoted'"]) == "bar"
     assert TestRepo.one(from o in Order, select: o.meta["';"]) == nil
     assert TestRepo.one(from o in Order, select: o.meta["\"double quoted\""]) == "baz"
+    assert TestRepo.one(from o in Order, select: o.meta["enabled"]) == true
```
Comment (leandrocp) [inline integration_test/cases/type.exs:321]: `== true` is redundant but trying to make it explicit if that makes sense

File: integration_test/cases/type.exs (hunk 1 above)
Comment (leandrocp) [reply integration_test/cases/type.exs]: Thanks! Decided to reuse `test "json_extract_path with primitive values"` instead of adding a new test case.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**